src/hnsw/build.c

```diff
@@ -385,6 +385,93 @@ static void FreeBuildState(HnswBuildState *buildstate)
     MemoryContextDelete(buildstate->tmpCtx);
 }
 
+/*
+ * This function will be called when an external file will be passed (_experimental_index_file_path param)
+ * we should check if user has also passed index params verify that it matches with the ones defined in index file
+ * header With this we ensure that on REINDEX operation, if we use that params to construct the index
+ * it will match
+ */
+static void ValidateIndexFileParams(Relation index, usearch_init_options_t *opts)
+{
+    ldb_HnswOptions *index_options = (ldb_HnswOptions *)index->rd_options;
+    assert(index_options != NULL);
+
+    if(index_options->ef && (size_t)index_options->ef != opts->expansion_search) {
+        elog(ERROR,
+             "Index option 'ef'(%u) does not match with the index file option 'ef'(%lu)",
```

Comment:
...does not match the 'ef' parameter with which the external file index was created